### DISCUSSION TITLE
How do I set the return type of the useFetch function?

### DISCUSSION THREAD

**Discussion by zestlee1106** [nuxt/nuxt#19497] [Questions]
I used to use it as below.

```typescript
type User = {
 username: string
 password: string
}

const user = useFetch<User>('/user')
```

However, after updating to version 3.2.2 the return type is not applied.

How do I set the return type of the useFetch function?

**Comment by imlautaro** [comment]
The behavior with useFetch typing hasn't changed, but useFetch returns a promise. You should await for the result.

For example, in your case it should be something like:
```typescript
interface User {
  username: string
  password: string
}

const { data: user } = await useFetch<User>('/user')
```

**Reply by zestlee1106** [reply]
I found a problem.

An error occurred in setting the option for useFetch.

Thanks for your help.

**Reply by hyungkishin** [reply]
진짜 멋있어요
